example/src/main/java/com/stripe/example/view/CreditCardView.java

```diff
@@ -0,0 +1,560 @@
+                    }
+                }
+                return false;
+            }
+        });
+
+        mEtExpiryDate.addTextChangedListener(new TextWatcher() {
+            boolean freeze = false;
+            boolean hadSlash = false;
+
+            @Override
+            public void beforeTextChanged(CharSequence charSequence, int start, int count, int after) {
+                if (!freeze) {
+                    hadSlash = mEtExpiryDate.getText().toString().contains("/");
+                }
+            }
+
+            @Override
+            public void onTextChanged(CharSequence charSequence, int start, int before, int count) {
+                if (!freeze) {
+                    String curString = mEtExpiryDate.getText().toString();
+
+                    if (curString.isEmpty()) {
+                        return;
+                    }
+
+                    if (hadSlash && !mEtExpiryDate.getText().toString().contains("/")) {
+                        freeze = true;
+                        mEtExpiryDate.setText("");
+                        freeze = false;
+                        return;
+                    }
+
+                    mEtExpiryDate.setTextColor(mColor);
+
+                    String monthStr = null;
+                    String yearStr = null;
+
+                    String []parts = curString.split("[/]");
+                    if (parts.length > 0 && !StripeTextUtils.isBlank(parts[0])) {
+                        try {
+                            monthStr = parts[0].substring(0, parts[0].length() > 2 ? 2 : parts[0].length());
+                            int month = Integer.parseInt(monthStr);
+                            mCard.setExpMonth(month);
+                            if (parts[0].length() == 1 && month > 1) {
+                                monthStr = String.format("%02d", month);
+                            }
+                        } catch (NumberFormatException e) {
+                            e.printStackTrace();
+                            freeze = true;
+                            mEtExpiryDate.setText("");
+                            freeze = false;
+                            return;
+                        }
+                        if (mCard.validateExpMonth() && parts.length > 1) {
+                            try {
+                                yearStr = parts[1].substring(0, parts[1].length() > 2 ? 2 : parts[1].length());
+                                int year = Integer.parseInt(yearStr);
+                                mCard.setExpYear(year);
+                            } catch (NumberFormatException e) {
+                                e.printStackTrace();
+                                freeze = true;
+                                mEtExpiryDate.setText("");
+                                freeze = false;
+                                return;
+                            }
+                        }
+                    }
+
+                    freeze = true;
```

Comment:
The difference (I think) is whether or not you want the method to still get called when the text watcher causes a text change, then make sure not to hit this block that you have separated out with the `freeze` variable (or stop right above it), or if you don't mind it being called over and over and just skip the big if block (particularly if you want anything else to happen).

It might be fine to leave as is -- it certainly works, and it's not like this is causing looping.